ml-algorithms/src/main/java/org/opensearch/ml/engine/httpclient/MLHttpClientFactory.java

```diff
@@ -8,79 +8,63 @@
 import java.net.Inet4Address;
 import java.net.InetAddress;
 import java.net.UnknownHostException;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
 import java.util.Arrays;
+import java.util.Locale;
+import java.util.Optional;
 
 import org.apache.commons.lang3.math.NumberUtils;
-import org.apache.http.HttpHost;
-import org.apache.http.HttpRequest;
-import org.apache.http.HttpResponse;
-import org.apache.http.conn.UnsupportedSchemeException;
-import org.apache.http.impl.client.CloseableHttpClient;
-import org.apache.http.impl.client.HttpClientBuilder;
-import org.apache.http.impl.client.LaxRedirectStrategy;
-import org.apache.http.impl.conn.DefaultSchemePortResolver;
-import org.apache.http.protocol.HttpContext;
-
-import com.google.common.annotations.VisibleForTesting;
 
 import lombok.extern.log4j.Log4j2;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
 
 @Log4j2
 public class MLHttpClientFactory {
 
-    public static CloseableHttpClient getCloseableHttpClient() {
-        return createHttpClient();
+    public static SdkAsyncHttpClient getAsyncHttpClient() {
+        try {
+            return AccessController
+                .doPrivileged(
+                    (PrivilegedExceptionAction<SdkAsyncHttpClient>) () -> NettyNioAsyncHttpClient.builder().maxConcurrency(100).build()
+                );
+        } catch (PrivilegedActionException e) {
+            return null;
```

Comment:
what happens when we return null? Add test?

Reply:
The inference request will return MLException here: https://github.com/opensearch-project/ml-commons/pull/1958/files#diff-6c364bbd919d4840502401e3ada03134cd3c3f71571cbd458c4e2a3aea0da745R87 and https://github.com/opensearch-project/ml-commons/pull/1958/files#diff-dbff9b113bf7a6cdba66b054415ce4f40be8499f4164b7b2c72de815294c81e9R102. It's not easy to add UT for this since we don't have a convenient way to mock static methods.

Reply:
[isSuperAdminUser](https://github.com/opensearch-project/ml-commons/blob/0903d5da4bc9fb8051621de05759dbdd36613972/plugin/src/main/java/org/opensearch/ml/utils/RestActionUtils.java#L228) is an static method. Corresponding test: https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/test/java/org/opensearch/ml/utils/RestActionUtilsTests.java#L294-L310


While using this static method to other class, I created a stub ([example](https://github.com/opensearch-project/ml-commons/blob/0903d5da4bc9fb8051621de05759dbdd36613972/plugin/src/main/java/org/opensearch/ml/action/models/GetModelTransportAction.java#L162-L166)), Now we can use that stubbing method in test. ([example](https://github.com/opensearch-project/ml-commons/blob/0903d5da4bc9fb8051621de05759dbdd36613972/plugin/src/test/java/org/opensearch/ml/action/models/GetModelTransportActionTests.java#L151))

Reply:
Added tests for this case.